source/tests/system/device_server.cpp

```diff
@@ -0,0 +1,86 @@
+#include "device_server.h"
+
+#include <server/device_enumerator.h>
+#include <server/session_utilities_service.h>
+#include <server/syscfg_library.h>
+#include <niscope/niscope_library.h>
+#include <niscope/niscope_service.h>
+#include <niswitch/niswitch_library.h>
+#include <niswitch/niswitch_service.h>
+#include <nisync/nisync_library.h>
+#include <nisync/nisync_service.h>
+#include <nidcpower/nidcpower_library.h>
+#include <nidcpower/nidcpower_service.h>
+
+namespace ni {
+namespace tests {
+namespace system {
+
+DeviceServerInterface::~DeviceServerInterface() {}
+
+class DeviceServer : public DeviceServerInterface {
+ public:
+  DeviceServer();
+  ~DeviceServer() override {};
+
+  // DeviceServerInterface overrides
+  std::shared_ptr<::grpc::Channel> InProcessChannel() override;
+
+ private:
+  nidevice_grpc::SessionRepository session_repository_;

```

Comment:
I'm a little skeptical of keeping a static singleton session repository around for all tests to share. If one service messes up the session repository it has the potential of impacting the results of other tests. I'm not too worried about the services themselves since they don't keep state apart from the loaded function pointers. Maybe it would be sufficient just to have tests call reset on the repository from init.

Reply:
I updated the tests to reset session_repository_ on test setup (i.e. in the test fixture constructor).